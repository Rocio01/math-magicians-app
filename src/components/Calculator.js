import React from 'react';
// eslint-disable-next-line react/prefer-stateless-function
export default class Calculator extends React.PureComponent {
  render() {
    return (
      <div className="calculator row">
        <div className="result float-right">
          <p className="py-1">0</p>
        </div>
        <div className="calculator-buttons row">
          <div className="calculator-digits container col row">
            <button type="button" className=" col-4">AC</button>
            <button type="button" className=" col-4">+/-</button>
            <button type="button" className=" col-4">%</button>
            <button type="button" className=" col-4">7</button>
            <button type="button" className=" col-4">8</button>
            <button type="button" className=" col-4">9</button>
            <button type="button" className=" col-4">4</button>
            <button type="button" className=" col-4">5</button>
            <button type="button" className=" col-4">6</button>
            <button type="button" className=" col-4">1</button>
            <button type="button" className=" col-4">2</button>
            <button type="button" className=" col-4">3</button>
            <button type="button" className=" col-10">0</button>
            <button type="button" className=" col-2">.</button>

          </div>
          <div className="calculator-operators container col-2 row">
            <button type="button">/</button>
            <button type="button">x</button>
            <button type="button">-</button>
            <button type="button">+</button>
            <button type="button">=</button>
          </div>

        </div>
      </div>
    );
  }
}
