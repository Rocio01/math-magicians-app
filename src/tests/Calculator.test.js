import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('matches calculator snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
