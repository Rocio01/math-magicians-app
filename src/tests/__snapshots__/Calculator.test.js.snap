// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches calculator snapshot 1`] = `
<div
  className="row"
>
  <h2
    className="p-2 col-4"
  >
     lets do a little math!
  </h2>
  <div
    className="col"
  >
    <div
      className="calculator row align-items-center"
    >
      <div
        className="result float-right"
      >
        <p
          className="py-3"
        >
          0
        </p>
      </div>
      <div
        className="calculator-buttons row"
      >
        <div
          className="calculator-digits container col row"
        >
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className=" col-4"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className=" col-9"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className=" col-3"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </div>
        <div
          className="calculator-operators container col-3 row"
        >
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
